sematic/db/migrations/20230609124213_pre_alembic_parity.sql

```diff
@@ -0,0 +1,47 @@
+-- migrate:up
+
+UPDATE runs SET function_path = '' WHERE function_path IS NULL;
+
+ALTER TABLE artifacts ALTER COLUMN type_serialization SET NOT NULL;
+ALTER TABLE edges ALTER COLUMN artifact_id SET NOT NULL;
+ALTER TABLE jobs ALTER COLUMN created_at SET NOT NULL;
+ALTER TABLE jobs ALTER COLUMN updated_at SET NOT NULL;
+ALTER TABLE notes ALTER COLUMN created_at SET NOT NULL;
+ALTER TABLE notes ALTER COLUMN updated_at SET NOT NULL;
+ALTER TABLE runs ALTER COLUMN function_path SET NOT NULL;
+ALTER TABLE runs ALTER COLUMN root_id SET NOT NULL;
+ALTER TABLE runs ALTER COLUMN tags SET NOT NULL;
+ALTER TABLE runs ALTER COLUMN source_code SET NOT NULL;
+
+ALTER TABLE edges ADD CONSTRAINT edges_destination_run_id_fkey FOREIGN KEY (destination_run_id) REFERENCES runs(id);
```

Comment:
these were mistakenly pointing to `artfacts(id)` instead of `runs(id)` in the sqlalchemy code (see below), and the db itself did not have any constraint at all, so i've now added the _corrected_ constraint here and fixed the sqlalchemy bug below.